backend/apps/readings/analysis.py

```diff
@@ -370,3 +370,114 @@ def compute_median_view_time(self):
             list_of_times.sort()
             median_view_time = statistics.median(list_of_times)
         return round(median_view_time)
+
+    def compute_mean_response_length(self):
+        """
+        Given a list of student response dicts,
+        return the mean character length (across all users) of the response
+        :return: float, mean number of characters in the user's response
+        """
+        list_of_responses = [r.response for r in self.responses]
+        mean_response_length = 0
+        for response in list_of_responses:
+            mean_response_length += len(response)
+        return round(mean_response_length / len(list_of_responses), 2)
+
+    def get_number_of_unique_students(self):
```

Comment:
You can do this with a Django queryset filter instead, so it happens at db query time. See https://docs.djangoproject.com/en/2.2/ref/models/querysets/#django.db.models.query.QuerySet.distinct